Add back editor-post-locked-modal to post lock component.

spacedmonkey opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?

The `editor-post-locked-modal` class name on the post lock component was remove in
https://github.com/WordPress/gutenberg/pull/58197

This PR brings back this class name. 

## Why?

This class name is useful and my team uses to customise the post locking dialog. We use it for style the component, add extra buttons and for e2e tests. 

Also it means that classes like `editor-post-locked-modal__avatar` do not make sense. `editor-post-locked-modal__avatar` is meant a sub class of `editor-post-locked-modal` that no longer exists. 

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->

### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->

## Screenshots or screencast <!-- if applicable -->

spacedmonkey committed:
Add back editor-post-locked-modal to post lock component.

Mamaduka reviewed (approved):
Makes sense ✅

tyxla reviewed (approved):
Thanks @spacedmonkey 🙌

youknowriad commented on packages/editor/src/components/post-locked-modal/index.js:
```diff
@@ -178,6 +178,7 @@ export default function PostLockedModal() {
 			shouldCloseOnClickOutside={ false }
 			shouldCloseOnEsc={ false }
 			isDismissible={ false }
+			className="editor-post-locked-modal"
```
Can you add a comment explaining that this is here because it used by plugins.

Also worth nothing that classnames are not public APIs officially but we can keep them around if it doesn't cost core anything in terms of complexity/performance.

jonnynews replied:
Fixed in https://github.com/WordPress/gutenberg/pull/64257/commits/14e9eb65afad4034fff3be409a69d2e94eabcf1b. 

@youknowriad What is the recommend way to extend or override this component if not using this class? At the moment, I can't see a way.

spacedmonkey replied:
Fixed in https://github.com/WordPress/gutenberg/commit/14e9eb65afad4034fff3be409a69d2e94eabcf1b.

@youknowriad What is the recommend way to extend or override this component if not using this class? At the moment, I can't see a way.

youknowriad replied:
There is no official way at the moment. In general it might be advised to raise an issue to discuss the use-case (and related use-cases) to see if we can come up with a good API.

spacedmonkey committed:
Add comment to preserve class in post-locked-modal

Ensure the `editor-post-locked-modal` class remains unchanged because third-party plugins depend on it. This change prevents potential breakages in plugin functionality.
